pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/RequestProcessor.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.compat.tests;
+
+import java.io.File;
+import java.net.URI;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpVersion;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpDelete;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.mime.HttpMultipartMode;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.entity.mime.content.ContentBody;
+import org.apache.http.entity.mime.content.FileBody;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.apache.pinot.spi.utils.JsonUtils;
+
+
+public class RequestProcessor {
```

Comment:
I somehow feel we should wrapper something in pinot-client for management APIs. So both users and devs can reuse the code. 

@mayankshriv @mcvsubbu @Jackie-Jiang

Reply:
If we do expose management APIs, then instead of using the same jar, I feel like adding another jar is better. The management API may have more dependencies on other internal things that we may not want pinot-client to depend on.

But then most admins will not be coding in java, and may be just fine with a well-defined REST API. So, a management API may not add much value.

Reply:
In general, I tend to confirm with DRY principle. There can be modularization as well as packaging to expose the right library to the right audience. Whether it is worth it in this particular case (especially if it increases the scope of the changes drastically), I am unsure. My $0.02.

Reply:
I don't think (or, at least, I hope not) @fx19880617 is proposing to do it as a part of the testing scope. I think it was a general question. We should take that elsewhere, since it definitely does not fit into this scope (of compat tests)

Reply:
Yep, and that's why my response started with "In general" :-)